python/pyarrow/_flight.pyx

```diff
@@ -177,6 +185,16 @@ cdef class FlightUnavailableError(FlightError, ArrowException):
         return MakeFlightError(CFlightStatusUnavailable, tobytes(str(self)),
                                self.extra_info)
 
+
+class FlightWriteSizeExceededError(OSError, ArrowException):
```

Comment:
I'm not sure I get the point of `OSError` here. I would expect `ArrowInvalid`.

Reply:
Ah, I did this because it originates as IOError (which gets mapped to OSError in Python), but I've changed both to ArrowInvalid.